app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java

```diff
@@ -301,70 +280,52 @@ public boolean onContextItemSelected(MenuItem item) {
-                R.layout.all_episodes_fragment);
-    }
-
-    View onCreateViewHelper(LayoutInflater inflater,
-                            ViewGroup container,
-                            Bundle savedInstanceState,
-                            int fragmentResource) {
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         super.onCreateView(inflater, container, savedInstanceState);
+        View root = inflater.inflate(R.layout.all_episodes_fragment, container, false);
 
-        View root = inflater.inflate(fragmentResource, container, false);
+        MainActivity mainActivity = (MainActivity) getActivity();
+        listAdapter = new AllEpisodesRecycleAdapter(mainActivity, itemAccess, showOnlyNewEpisodes());
+        listAdapter.setHasStableIds(true);
 
-        recyclerView = root.findViewById(android.R.id.list);
-        RecyclerView.ItemAnimator animator = recyclerView.getItemAnimator();
-        if (animator instanceof SimpleItemAnimator) {
-            ((SimpleItemAnimator) animator).setSupportsChangeAnimations(false);
-        }
         layoutManager = new LinearLayoutManager(getActivity());
+        recyclerView = root.findViewById(android.R.id.list);
         recyclerView.setLayoutManager(layoutManager);
         recyclerView.setHasFixedSize(true);
         recyclerView.addItemDecoration(new HorizontalDividerItemDecoration.Builder(getActivity()).build());
+        recyclerView.setVisibility(View.GONE);
+        recyclerView.setAdapter(listAdapter);
 
-        progLoading = root.findViewById(R.id.progLoading);
-
-        if (!itemsLoaded) {
-            progLoading.setVisibility(View.VISIBLE);
+        RecyclerView.ItemAnimator animator = recyclerView.getItemAnimator();
+        if (animator instanceof SimpleItemAnimator) {
+            ((SimpleItemAnimator) animator).setSupportsChangeAnimations(false);
         }
 
-        viewsCreated = true;
-
-        if (itemsLoaded) {
-            onFragmentLoaded();
-        }
+        progLoading = root.findViewById(R.id.progLoading);
+        progLoading.setVisibility(View.VISIBLE);
 
         emptyView = new EmptyViewHandler(getContext());
         emptyView.attachToRecyclerView(recyclerView);
         emptyView.setTitle(R.string.no_all_episodes_head_label);
         emptyView.setMessage(R.string.no_all_episodes_label);
+        emptyView.hide();
 
         return root;
     }
 
     private void onFragmentLoaded() {
         if (episodes != null && episodes.size() > 0) {
-            if (listAdapter == null) {
-                MainActivity mainActivity = (MainActivity) getActivity();
-                listAdapter = new AllEpisodesRecycleAdapter(mainActivity, itemAccess, showOnlyNewEpisodes());
-                listAdapter.setHasStableIds(true);
-                recyclerView.setAdapter(listAdapter);
-                emptyView.updateAdapter(listAdapter);
-            }
```

Comment:
Oh, that's odd. I'll put it back, then.

Reply:
Recreating the adapter every time breaks animations. It is only needed if it had been empty before.

Reply:
@andersonvom Running this PR since a few days and did not notice any problems (except for the missing animations when the adapter is recreated too often)

Reply:
That's great! I just pushed the changes to only recreate the adapter when needed (i.e. when the item count gets to zero)